### PR TITLE
Add model tests

### LINKED ISSUE #10: Add tests for the Role, TeamType, Member, Team, and Event models
We need to write tests for the Member, Team, Event, TeamType, and Rol models that test out the requirements defined by each Schema. These requirements include...
- if a field is required
- if a field has a default value
- if a field has a given type

These tests will serve two purposes:

1. They need to exist anyway to make sure we are careful about future changes.

2. They will give future developers an idea about how to write tests themselves, thereby encouraging them to write tests for their PRs :smile: 

Each model should have its own test file in the `tests` directory.

### PR DESCRIPTION
**Pull request by karpawich**
Resolves #10

### CODE REVIEW

File: server/models/role.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+Role.isRole = function isRole(string) {
+  return Object.values(Role).indexOf(string) !== -1;
+};
+
+/**
+   * Returns whether a member with the given role is at the given permission level.
+   * @param {Role} role The role of the member
+   * @param {Role} permissionLevel The permission level
+   */
+Role.hasPermission = function hasPermission(role, permissionLevel) {
```
Comment (Arthelon) [inline server/models/role.js]: Is it necessary to have a permission level in addition to role? I would think your permission level is contained within the role.

Comment (Arthelon) [reply server/models/role.js]: Also, I think this method should be part of the `Member` model. Roles are tied to Members so we can abstract this away.

File: test/team.model.test.js [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    it("should be a TeamType enum value", async function () {
+      team = Team({ name: "exec", type: "abcdef" });
+      try {
+        await team.save();
+      } catch (e) {
+        err = e;
+      }
+      expect(err).to.be.instanceof(Error);
+      expect(err).to.have.property("name", "ValidationError");
+      expect(err).to.have.property("message", "Team validation failed: type: \"abcdef\" is not a value of the TeamType enum");
```
Comment (Arthelon) [inline test/team.model.test.js]: add additional code and assertions for when the `type` is assigned a valid `TeamType`

File: test/member.model.test.js [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+      });
+      try {
+        await member.save();
+      } catch (e) {
+        err = e;
+      }
+      expect(err).to.be.instanceof(Error);
+      expect(err).to.have.property("name", "ValidationError");
+      expect(err).to.have.property("message", "Member validation failed: role: Cast to string failed for value \"{}\" at path \"role\"");
+    });
```
Comment (Arthelon) [inline test/member.model.test.js:383]: Add additional test cases to check for valid/invalid `role` properties

**Review by Arthelon — CHANGES_REQUESTED** [review]
Tests look good overall. Some higher-level feedback:
- Instead of establishing a DB connection, we should use something like [this](https://github.com/nodkz/mongodb-memory-server) to mock the DB connection so that we don't accidentally dump a production DB and so that tests run faster.
- `Role` and `TeamType` enums are very coupled with the `Member` and `Team` models. I don't think they should be considered models and we should move as much of the validation/extra logic attached to those enums into the models. We can just expose the enums as an export from these models. (i.e. `export { Role } from 'models/member'`)

File: server/models/role.js (hunk 1 above)
Comment (karpawich) [reply server/models/role.js]: Good point. It'll be even more confusing for people who contribute later on, so I just ended up removing it.

**Comment by karpawich** [comment]
All requested changes are now resolved! Merge time

**Pull request merged**